src/consul_configs.rs

```diff
@@ -5,71 +5,76 @@ use failure::Error;
 use serde_json;
 use std::{thread, time};
 
-pub fn wait_or_get(
-    chain: &str,
-    key: &str,
-    consul_url: &str,
-    consul_token: &str,
-    sidechain_name: &str,
-) -> Result<String, Error> {
-    let keyname = format!("chain/{}/{}", &sidechain_name, &chain);
-    let first = Box::new(true);
-    let json = consul_select(keyname.as_ref(), consul_url, consul_token, || {
-        if *first {
-            info!("chain for config not available in consul yet");
+pub struct ConsulConfig {
+    consul_url: String,
+    consul_token: String,
+    sidechain_name: String,
+}
+
+impl ConsulConfig {
+    pub fn new(consul_url: &str, consul_token: &str, sidechain_name: &str) -> Self {
+        Self {
+            consul_url: consul_url.to_string(),
+            consul_token: consul_token.to_string(),
+            sidechain_name: sidechain_name.to_string(),
         }
-    })?;
+    }
 
-    info!("chain for {:?} config available in consul now", chain);
+    pub fn wait_or_get(&self, chain: &str, key: &str) -> Result<String, Error> {
+        let keyname = format!("chain/{}/{}", &self.sidechain_name, &chain);
+        let first = Box::new(true);
+        let json = self.consul_select(keyname.as_ref(), || {
+            if *first {
+                info!("chain for config not available in consul yet");
+            }
+        })?;
+        info!("chain for {:?} config available in consul now", chain);
 
-    json[&key]
-        .as_str()
-        .map_or(Err(OperationError::CouldNotGetConsulKey(key.to_string()).into()), |v| {
-            Ok(v.to_string())
-        })
-}
+        if json[&key].is_u64() {
```

Comment:
I think it'd be better to return a serde_json Value here and have the caller deal with the type, or figure out some way to make this generic vs special casing types.

This whole module kinda needs some love though imo so im not gonna have this block a merge, ref #13

Reply:
I agree. It would probably be easier to return a serde_json value too. Going to merge this for now and add more improvements in https://github.com/polyswarm/relay/pull/47.